Utilities/DataLoading/EdFi.LoadTools.Test/SmokeTests/GetVersionTests.cs

```diff
@@ -6,37 +6,56 @@
 using System.Configuration;
 using EdFi.LoadTools.Engine;
 using EdFi.LoadTools.SmokeTest.CommonTests;
-using Microsoft.Owin.Hosting;
 using NUnit.Framework;
 using Moq;
-using Owin;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.AspNetCore.TestHost;
+using System.Net.Http;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.Hosting;
 
 namespace EdFi.LoadTools.Test.SmokeTests
 {
     [TestFixture]
     public class GetVersionTests
     {
         [Test]
-        public void Should_succeed_against_a_running_server()
+        public async System.Threading.Tasks.Task Should_succeed_against_a_running_serverAsync()
         {
-            var address = ConfigurationManager.AppSettings["TestingWebServerAddress"];
+            //TODO should be replaced with json
+            // var address = ConfigurationManager.AppSettings["TestingWebServerAddress"];
+            var address = "http://localhost:23456/";
 
-            using (WebApp.Start(
-                address, app => { app.Run(async context => { await context.Response.WriteAsync("successful result"); }); }))
-            {
-                var configuration = Mock.Of<IApiMetadataConfiguration>(cfg => cfg.Url == address);
-                var subject = new GetStaticVersionTest(configuration);
-                var result = subject.PerformTest().Result;
-                Assert.IsTrue(result);
-            }
+            var hostBuilder = new HostBuilder()
+                .ConfigureWebHost(
+                    webHost =>
+                    {
+                        // Add TestServer
+                        webHost.UseTestServer();
+
+                        webHost.Configure(
+                            app => app.Run(
+                                async ctx =>
+                                    await ctx.Response.WriteAsync("successful result")));
+                    });
+
+            var host = await hostBuilder.StartAsync();
+            var client = host.GetTestClient();
+            client.BaseAddress = new System.Uri(address);
+
+            var configuration = Mock.Of<IApiMetadataConfiguration>(cfg => cfg.Url == address);
+            var subject = new GetStaticVersionTest(configuration, client);
+            var result = subject.PerformTest().Result;
```

Comment:
this should be await subject.PerformTest()

Reply:
Updated